src/generators/android/android.ts

```diff
@@ -0,0 +1,328 @@
+import { camelCase, upperFirst } from 'lodash'
+import { Type, Schema, getPropertiesSchema } from '../ast'
+import * as Handlebars from 'handlebars'
+import { Generator, GeneratorClient, BasePropertyContext } from '../gen'
+
+// These contexts are what will be passed to Handlebars to perform rendering.
+// Everything in these contexts should be properly sanitized.
+
+interface AndroidObjectContext {
+	// The formatted name for this object, ex: "ProductClicked"
+	name: string
+}
+
+interface AndroidPropertyContext {
+	// The formatted name for this property, ex: "numAvocados".
+	name: string
+	// The type of this property. ex: "String".
+	type: string
+	// Stringified property modifiers. ex: "final, @Nonnull".
+	modifiers: Modifier
+	// Whether the property is nullable (@Nonnull vs @Nullable modifier).
+	isVariableNullable: boolean
+	// Whether null is a valid value for this property when sent to Segment.
+	nonNullPayload: boolean
+}
+
+interface AndroidTrackCallContext {
+	// The formatted function name, ex: "orderCompleted".
+	functionName: string
+	propsParam: boolean
+}
+
+enum JavaType {
+	String = 'String',
+	Long = 'Long',
+	Double = 'Double',
+	Boolean = 'Boolean',
+	Object = 'Object',
+}
+
+export const android: Generator<
+	{},
+	AndroidTrackCallContext,
+	AndroidObjectContext,
+	AndroidPropertyContext
+> = {
+	generatePropertiesObject: true,
+	namer: {
+		// See: https://github.com/AnanthaRajuCprojects/Reserved-Key-Words-list-of-various-programming-languages/blob/master/Java%20Keywords%20List.md
+		// prettier-ignore
+		reservedWords: [
+      "abstract", "assert", "boolean", "break", "byte", "case", "catch", "char", "class", "const",
+      "continue", "default", "do", "double", "else", "enum", "extends", "final", "finally", "float",
+      "for", "goto", "if", "implement", "imports", "instanceof", "int", "interface", "long", "native",
+      "new", "package", "private", "protected", "public", "return", "short", "static", "strictfp", "super",
+      "switch", "synchronized", "this", "throw", "throws", "transient", "try", "void", "volatile", "while"
+    ],
+		quoteChar: '"',
+		allowedIdentifierStartingChars: 'A-Za-z_$',
```

Comment:
TIL that `$` is a valid Java identifier.. For typewriter, let's avoid `$` -- feel free to remove it from both here :)